python/kserve/kserve/storage.py

```diff
@@ -117,16 +117,32 @@ def _download_s3(uri, temp_dir: str):
                 continue
             # In the case where bucket_path points to a single object, set the target key to bucket_path
             # Otherwise, remove the bucket_path prefix, strip any extra slashes, then prepend the target_dir
+            # Example:
+            # s3://test-bucket
+            # Objects: /a/b/c/model.bin /a/model.bin /model.bin
+            #
+            # If 'uri' is set to "s3://test-bucket", then the dowloader will
+            # download all the objects listed above, re-creating their subpaths
+            # under the temp_dir.
+            # If 'uri' is set to "s3://test-bucket/a", then the downloader will
+            # add to temp_dir: b/c/model.bin and model.bin.
+            # If 'uri' is set to "s3://test-bucket/a/b/c/model.bin", then
+            # the downloader will add to temp dir: model.bin
+            # (without any subpaths).
             target_key = (
-                obj.key
+                obj.key.rsplit("/", 1)[1]
```

Comment:
would this be an issue when `storageUri` is `s3://test-bucket/model.bin` and `obj.key` is `model.bin`?

Reply:
@yuzisun good point! I replaced the [1] with [-1], it should work fine even when the rsplit list is one element:
```
>>> "model.bin".rsplit("/",1)[-1]
'model.bin'
>>> "/a/b/c/model.bin".rsplit("/",1)[-1]
'model.bin'
```

Added also a specific unit test for it, lemme know :)